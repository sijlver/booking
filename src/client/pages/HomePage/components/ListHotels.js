import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import HotelItem from './HotelItem';

function ListHotels(props) {
    const { listHotels } = props;

    return (
        <div className='listHotels'>
            <ul>
                {
                    listHotels.map((hotel) => (
                        <HotelItem key={hotel.id} {...hotel} />
                    ))
                }
            </ul>
        </div>
    );
}

ListHotels.propTypes = {
    listHotels: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)),
};

export default ListHotels;
