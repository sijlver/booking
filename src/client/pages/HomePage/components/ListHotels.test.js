import React from 'react';
import { shallow } from 'enzyme';

import ListHotels from './ListHotels';

describe('ListHotels', () => {
    it('should render self', () => {
        const props = {
            listHotels: [{
                locationName: 'Hurghada, Egypt',
                id: '1',
                label: 'Steigenberger Al Dau Beach Hotel',
            }],
        };
        const wrapper = shallow(
            <ListHotels {...props} />,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
