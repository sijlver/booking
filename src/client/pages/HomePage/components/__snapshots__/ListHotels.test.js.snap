// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHotels should render self 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListHotels
    listHotels={
      Array [
        Object {
          "id": "1",
          "label": "Steigenberger Al Dau Beach Hotel",
          "locationName": "Hurghada, Egypt",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <HotelItem
          id="1"
          label="Steigenberger Al Dau Beach Hotel"
          locationName="Hurghada, Egypt"
        />
      </ul>,
      "className": "listHotels",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HotelItem
            id="1"
            label="Steigenberger Al Dau Beach Hotel"
            locationName="Hurghada, Egypt"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "id": "1",
            "label": "Steigenberger Al Dau Beach Hotel",
            "locationName": "Hurghada, Egypt",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <HotelItem
            id="1"
            label="Steigenberger Al Dau Beach Hotel"
            locationName="Hurghada, Egypt"
          />
        </ul>,
        "className": "listHotels",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HotelItem
              id="1"
              label="Steigenberger Al Dau Beach Hotel"
              locationName="Hurghada, Egypt"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "id": "1",
              "label": "Steigenberger Al Dau Beach Hotel",
              "locationName": "Hurghada, Egypt",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
